### PR TITLE
Modernize instructor assessment downloads

### PR DESCRIPTION
**Pull request by mylesw27**
This PR is part of the larger modernization effort. This PR intends to convert the instructorAssessmentDownloads controller to TypeScript and the page to HTML.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts [hunk 1]
```diff
@@ -57,18 +57,11 @@ router.get(
...
     }
     setFilenames(res.locals);
-    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    res.send(InstructorAssessmentDownloads({ resLocals: res.locals }));
   }),
 );
 
-/**
- * Local abstraction to adapt our internal notion of columns to the columns
- * format that the CSV `stringify()` function expects.
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts]: Added comment back in [7758962](https://github.com/PrairieLearn/PrairieLearn/pull/9483/commits/77589623346ff34dc97609b94602b9a34d3f9ac5).

File: apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.html.ts [hunk 2]
```diff
@@ -0,0 +1,431 @@
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+export function InstructorAssessmentDownloads({ resLocals }: { resLocals: Record<string, any> }) {
+  let identity = 'student';
+  if (resLocals.assessment.group_work) {
+    identity = 'group';
+  }
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.html.ts]: Updated locally in [7758962](https://github.com/PrairieLearn/PrairieLearn/pull/9483/commits/77589623346ff34dc97609b94602b9a34d3f9ac5).

File: apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts [hunk 3]
```diff
@@ -1,53 +1,57 @@
...
+import { InstructorAssessmentDownloads, Filenames } from './instructorAssessmentDownloads.html';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
-/** @typedef {[string, string][]} Columns */
+type Columns = [string, string][];
 
-const setFilenames = function (locals) {
+const getFilenames = function (locals) {
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts:18]: Nit:

```suggestion
function getFilenames(locals) {
```

File: apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts [hunk 4]
```diff
@@ -56,19 +60,18 @@ router.get(
...
-/**
+/*
  * Local abstraction to adapt our internal notion of columns to the columns
  * format that the CSV `stringify()` function expects.
- *
- * @param {[string, string][]} columns
- * @param {(record: any) => any} [transform]
  */
-function stringifyWithColumns(columns, transform = undefined) {
+
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts:73]: Nit, extraneous newline:

```suggestion
```

File: apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts [hunk 5]
```diff
@@ -57,18 +57,11 @@ router.get(
       throw error.make(403, 'Access denied (must be a student data viewer)');
     }
     setFilenames(res.locals);
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts]: Changed `setFilenames` to `getFilenames` and returned an object from it. That object is now being passed in to the HTML file. Also, I have updated the usage in the csv routes where we had previously been using `res.locals.filename` to now `filenames.filename`.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**